lib/inets/src/http_lib/http_uri.erl

```diff
@@ -81,13 +81,15 @@
 -type hex_uri() :: string() | binary(). %% Hexadecimal encoded URI.
 -type maybe_hex_uri() :: string() | binary(). %% A possibly hexadecimal encoded URI.
 
+-type scheme_defaults() :: [{scheme(), default_scheme_port_number()}].
+-type scheme_validation_fun() :: fun((SchemeStr :: string() | binary()) ->
+                                            valid | {error, Reason :: term()}).
```

Comment:
I see https://github.com/erlang/otp/blob/OTP-20.2.4/lib/inets/test/uri_SUITE.erl#L271-L274 says `non-fun scheme_validation_fun works as no option passed`. Though as a Developer I would expect incorrect value (non-fun) to cause some sort of error otherwise mistake in specification of value (e.g. mistake with some higher-order fun) could go unnoticed.

> So I propose to extend the type spec of scheme_validation_fun with the atom none.

I will follow as you propose in this PR.

Reply:
Addressed in 9fc5b13f91

(BTW if you want to squash before merge, you may enable GitHub to do that for you/us as an alternative to the classic merge.)